docs/templating_funcs.md

```diff
@@ -29,6 +29,31 @@ The `readFile` function allows you to read a file and return its content as the
 {{ $fileContent := readFile "./myfile" }}
 ```
 
+#### `readDir`
+the `readDir` function reads the files within provided directory path. (folders are excluded)
+
+```yaml
+{{ range $index,$item := readDir "./testdata/tmpl/sample_folder/" }}
+  {{- $itemSplit := splitList  "/" $item -}}
+  {{- if contains "\\" $item -}}
+  {{- $itemSplit = splitList "\\" $item -}}
+  {{- end -}}
+  {{- $itemValue := $itemSplit | last -}}
+  {{- $itemValue -}}
+{{- end -}}
+```
+
+#### `readDirEntries`
+The `readDirEntries` Returns a list of [https://pkg.go.dev/os#DirEntry](DirEntry) within provided directory path
```

Comment:
```suggestion
The `readDirEntries` returns a list of [https://pkg.go.dev/os#DirEntry](DirEntry) contained within the directory
```

I'm not a native English speaker/writer either but I guess perhaps this might be read easier 😁